charts/backstage/templates/sonataflows.yaml

```diff
@@ -0,0 +1,122 @@
+  {{- if (and (.Values.orchestrator.sonataflowPlatform.eventing.broker.name) (.Values.orchestrator.sonataflowPlatform.eventing.broker.namespace)) }}
+  eventing:
+    broker:
+      ref:
+        apiVersion: eventing.knative.dev/v1
+        kind: Broker
+        name: {{ .Values.orchestrator.sonataflowPlatform.eventing.broker.name }}
+        namespace: {{ .Values.orchestrator.sonataflowPlatform.eventing.broker.namespace }}
+  {{- end }}
+  services:
+    dataIndex:
+      enabled: true
+      persistence:
+        postgresql:
+          {{- if .Values.upstream.postgresql.enabled }}
+          secretRef:
+            name: {{ .Release.Name }}-postgresql-svcbind-postgres
+            userKey: username
+            passwordKey: password
+          serviceRef:
+            name: {{ .Release.Name }}-postgresql
+            namespace: {{ .Release.Namespace }}
+            databaseName: sonataflow
+          {{- else }}
+          secretRef:
+            name: {{ .Values.orchestrator.sonataflowPlatform.externalDBsecretRef }}
+            userKey: POSTGRES_USER
+            passwordKey: POSTGRES_PASSWORD
+          jdbcUrl: jdbc:postgresql://{{ .Values.upstream.backstage.appConfig.backend.database.connection.host}}:{{ .Values.upstream.backstage.appConfig.backend.database.connection.port}}/sonataflow?currentSchema=data-index-service
+          {{- end }}
+      {{- if .Values.orchestrator.sonataflowPlatform.dataIndexImage }}
+      podTemplate:
+        container:
+          image: {{ .Values.orchestrator.sonataflowPlatform.dataIndexImage }}
+      {{- end }}
+    jobService:
+      enabled: true
+      persistence:
+        postgresql:
+          {{- if .Values.upstream.postgresql.enabled }}
+          secretRef:
+            name: {{ .Release.Name }}-postgresql-svcbind-postgres
+            userKey: username
+            passwordKey: password
+          serviceRef:
+            name: {{ .Release.Name }}-postgresql
+            namespace: {{ .Release.Namespace }}
+            databaseName: sonataflow
+          {{- else }}
+          secretRef:
+            name: {{ .Values.orchestrator.sonataflowPlatform.externalDBsecretRef}}
+            userKey: POSTGRES_USER
+            passwordKey: POSTGRES_PASSWORD
+          jdbcUrl: jdbc:postgresql://{{ .Values.upstream.backstage.appConfig.backend.database.connection.host}}:{{ .Values.upstream.backstage.appConfig.backend.database.connection.port}}/sonataflow?currentSchema=jobs-service
+          {{- end }}
+      {{- if .Values.orchestrator.sonataflowPlatform.jobServiceImage }}
+      podTemplate:
+        container:
+          image: {{ .Values.orchestrator.sonataflowPlatform.jobServiceImage }}
+      {{- end }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-create-sonataflow-database
+  namespace: {{ .Release.Namespace }}
+spec:
+  template:
+    spec:
+      containers:
+      - name: psql
+        image: postgres:15
```

Comment:
With the airgap use case in mind, I think the image should be made configurable as well.